cmake/custom/examples.cmake

```diff
@@ -1,3 +1,10 @@
+#.rst:
+#
+# autocmake.yml configuration::
+#
+#   docopt: "--enable-examples Enable code examples [default: False]."
+#   define: "'-DENABLE_EXAMPLES={0}'.format(arguments['--enable-examples'])"
+
```

Comment:
This variable might be unbound/undefined in case a user decides to ditch the `setup` script.
You should try is to configure the project _without_ using the `setup` script, for example:
`cmake -H. -Bbuild` will configure the project to be built into the `build` directory. Then all unbound/undefined variables will cause trouble or lead to nonsensical configuration. In the latest Autocmake we added a couple of macros to remedy this issue. Here you should add a:
```
option_with_print(ENABLE_EXAMPLES "Enable code examples" OFF)
```
which will set it to off if `ENABLE_EXAMPLES` was not specified, ensuring that the variable is always in a well-defined state.

Reply:
Aha, is this also the reason this option does not show up in the cmake gui?

Reply:
Yep. It is never declared as an `option`.